multidim_threshold/projection.py

```diff
@@ -35,26 +36,37 @@ def generate_boundary_approxes(lo, hi, member_oracles, **kwargs):
         yield boundaries
 
 
-def generate_projections(lo, hi, member_oracles, *, direc=None, searches=None):
-    proj_vecs = generate_proj_vecs(lo, hi, direc)
+def generate_projections(lo, hi, member_oracles, *, direc=None, searches=None, random=False):
+    axis_hi = axes_intersects(lo, hi, member_oracles, random=False) if random else hi
```

Comment:
I think you want searches. I think you can swap this and lines 40-41 so that you can pass searches in directly